Prevent users from changing their own permissions

bartvanb opened the pull request:
As noted in Issue IKNL/vantage6-main#124, users are currently able to shrink their own permission space while this should not be the case. Therefore, several checks were added in this branch:
- in assigning roles or rules to a user via a patch request, this is no longer allowed by that user itself. This ensures users can neither shrink nor increase their permissions (increasing one's own permissions was already prevented by other checks as well) 
- deleting roles or deleting rules from a role can now no longer be done by a user that has this role

bartvanb committed:
Added checks so that user cannot increase or decrease their own permission space by assigning themselves different rules or roles

bartvanb committed:
Adding check that users cannot delete their own roles or rules from their own roles

frankcorneliusmartin commented on vantage6/server/resource/role.py:
```diff
@@ -227,6 +227,10 @@ def delete(self, id):
             return {"msg": f"Role with id={id} not found."}, \
                 HTTPStatus.NOT_FOUND
 
+        if role in g.user.roles:
```
A user *is* allowed to delete pre-defined roles. So this is not needed.  A user should not be able to remove a role from its profile.

However.. I dont think this endpoint is complete as associations to the role might still be present (e.g. users, rules)

frankcorneliusmartin replied:
I do understand that the user is still able to shrink his permission space in this way. But doing it this way it is more explicit that you are removing permissions.

bartvanb replied:
Agree that they are being explicit, reversing this change

frankcorneliusmartin commented on vantage6/server/resource/role.py:
```diff
@@ -315,6 +319,10 @@ def delete(self, id, rule_id):
             return {'msg': f'Rule id={rule_id} not found!'}, \
                 HTTPStatus.NOT_FOUND
 
+        if role in g.user.roles:
```
Same as comment as above.

bartvanb replied:
Same solution as above ;-)

bartvanb committed:
Removed extra checks on user identity in removing roles/rules using delete requests.

The user is being explicit in wanting to delete those roles/rules, so will be allowed to do so even if shrinking their own permission space
